### PR TITLE
feat: add optional defmt support

### PR DESCRIPTION
**Pull request by robamu**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -87,6 +87,7 @@ pub mod dht11 {
     use super::*;
 
     #[derive(Clone, Copy, Debug, PartialEq, Eq)]
+    #[cfg_attr(feature = "defmt", derive(defmt::Format))]
```
Comment (michaelbeaumont) [inline src/lib.rs:90]: Should this be declared in `Cargo.toml`?

Comment (robamu) [reply src/lib.rs:90]: The optional defmt dependency should implicitely add a defmt feature (https://doc.rust-lang.org/cargo/reference/features.html#optional-dependencies)

**Comment by robamu** [comment]
I just saw that the DHT22 has a separate reading structure where defmt can be added as well :)

File: src/lib.rs (hunk 1 above)
Comment (michaelbeaumont) [reply src/lib.rs:90]: ok!

**Review by michaelbeaumont — APPROVED** [review]

**Comment by michaelbeaumont** [comment]
Thanks!

**Pull request merged**